shared/AppInsightsCore/src/JavaScriptSDK/CookieMgr.ts

```diff
@@ -188,12 +287,20 @@ export function createCookieMgr(rootConfig?: IConfiguration, logger?: IDiagnosti
 
         _path = cookieMgrConfig.path || "/";
         _domain = cookieMgrConfig.domain;
+        
+        // Check if enabled state is changing
+        let wasEnabled = _enabled;
         // Explicitly checking against false, so that setting to undefined will === true
         _enabled = _isCfgEnabled(rootConfig, cookieMgrConfig) !== false;
 
         _getCookieFn = cookieMgrConfig.getCookie || _getCookieValue;
         _setCookieFn = cookieMgrConfig.setCookie || _setCookieValue;
         _delCookieFn = cookieMgrConfig.delCookie || _setCookieValue;
+
+        // If cookies were just enabled via config change and we have pending cookies, flush them
+        if (!wasEnabled && _enabled && !cookieMgrConfig.disableCaching) {
```

Comment:
create a local variable in the closure of the cookie managed called _disableCaching and set the value in the onConfigChange to reduce the size of the generated code, change all usages of this config to that variable.